lib/analytics/course_students_csv_builder.rb

```diff
@@ -17,6 +22,28 @@ def generate_csv
 
   private
 
+  def populate_created_articles
+    # A user has created an article during the course if
+    # the user is in the user_ids of new_articles_courses
+    # has a revision with new_article: true for that article
+    @course.new_articles_courses.pluck(:article_id, :user_ids).each do |article_id, user_ids|
+      user_ids.each do |user_id|
+        @created_articles[user_id] += 1 if article_creator?(article_id, user_id)
+      end
+    end
+  end
+
+  def article_creator?(article_id, user_id)
+    @revisions[article_id] == user_id
+  end
+
+  def populate_edited_articles
+    # A user has edited an article if the user is in the user_ids list of edited_articles_courses
+    @course.edited_articles_courses.pluck(:article_id, :user_ids).each do |_article_id, user_ids|
+      user_ids.each { |user_id| @edited_articles[user_id] += 1 }
+    end
+  end
```

Comment:
Similar to the populate edit articles but in this case the user gets a count for every line it's included.
I've decided to change to these populated methods because this way it iterates through the SQL result only once instead of once per user as it was being done previously.